Jasmine/Common/Models/TextGrid.swift

```diff
@@ -55,10 +60,8 @@ struct TextGrid {
     /// - Parameters:
     ///   - coord1: the first coordinate
     ///   - coord2: the second coordinate
-    mutating func swap(_ coord1: Coordinate, and coord2: Coordinate) {
-        let temp = grid[coord2.row][coord2.col]
-        grid[coord2.row][coord2.col] = grid[coord1.row][coord1.col]
-        grid[coord1.row][coord1.col] = temp
+    mutating func swap(coord1: Coordinate, coord2: Coordinate) {
+        Swift.swap(&self[coord1], &self[coord2])
```

Comment:
This one was made like this to not introduce concurrent writeback when using didSet over this `TextGrid`.

But anyways we can go this route, I'll change my grid VM